datafusion/physical-plan/src/joins/hash_join/stream.rs

```diff
@@ -650,6 +674,8 @@ impl HashJoinStream {
             )?
         };
 
+        self.batch_coalescer.push_batch(result)?;
```

Comment:
Maybe add a comment that ignoring `PushBatchStatus::LimitReached` result is intentional. Same at line 762 below.